client/src/utils/filtering.ts

```diff
@@ -323,12 +339,28 @@ export default class Filtering<T> {
      * @returns {Boolean} True if the filter is set to the given value
      * */
     checkFilter<T>(filterText: string, filterName: string, filterValue: T): boolean {
-        const re = new RegExp(`${filterName}:(\\S+)`);
-        const reMatch = re.exec(filterText);
-        const testValue = reMatch ? reMatch[1] : this.defaultFilters[filterName];
+        const testValue = this.getFilterValue(filterText, filterName);
         return toLowerNoQuotes(testValue) === toLowerNoQuotes(filterValue);
     }
 
+    /** Get the value of a particular filter from filterText.
+     * @param {String} filterText Raw filter text string
+     * @param {String} filterName Filter key to check
+     * @param {String} [alias="eq"] String alias for filter operator, e.g.:"lt"
+     * @returns {String | Boolean} The filterValue for the filter
```

Comment:
Remove duplicate (and conflicting string != String) type declarations.

```suggestion
     * @param filterText Raw filter text string
     * @param filterName Filter key to check
     * @param [alias="eq"] String alias for filter operator, e.g.:"lt"
     * @returns The filterValue for the filter
```

Reply:
Thanks! i will correct these, just confirming, there is no need to declare type in the documentation headers if we have type declaration in the function definition (like in line no. 352)

Reply:
That is correct. The typescript type declarations should fully cover type hinting.